src/components/menu-top/LoadModal.vue

```diff
@@ -76,18 +78,38 @@ export default Vue.extend({
       this.showPreview = null;
       this.parseError = "";
       const reader = new FileReader();
-      reader.onload = (): void => {
-        if (reader.result && reader.result instanceof ArrayBuffer) {
-          try {
-            this.showPreview = loadShowFromBuffer(reader.result);
-          } catch (e) {
-            this.parseError = e;
+      // Check if calchart 4 or not...
+      if (this.file.name.includes(".shw4")) {
+        reader.onload = (): void => {
+          if (reader.result) {
+            try {
+              if (reader.result)
+                this.showPreview = new Show(
+                  JSON.parse(reader.result as string)
+                );
+            } catch (e) {
+              this.parseError = e;
+            }
+          } else {
+            this.parseError = "Could not read file.";
           }
-        } else {
-          this.parseError = "Could not read file.";
-        }
-      };
-      reader.readAsArrayBuffer(this.file);
+        };
+        reader.readAsText(this.file);
+      } else {
```

Comment:
Could we change this to an `else if` to make sure that the file extension is `.shw`? If it is not `.shw` or `.shw4`, then we should show an error.